aip/general/0111.md

```diff
@@ -0,0 +1,90 @@
+---
+id: 111
+state: draft
+created: 2023-06-10
+placement:
+  category: api-concepts
+  order: 50
+---
+
+# Planes
+
+Resources and methods on an API can be divided into the *plane* that they reside
+or perform operations upon. For the context of APIs, the following planes are
+defined:
+
+- The management plane: a uniform, resource-oriented API that primarily
+  configures and allows retrieval of resources.
+- The data plane: a heteregenous API (ideally resource-oriented) that interface
+  with the resources that the the management plane configures.
+
+The term "plane" was originally used in networking architecture. Although system
+and network architecure often defines additional planes (e.g. control plane or
+power planes), as the AIPs are focused on the interface, they are not defined in
+this AIP.
+
+## Guidance
+
+### Management Plane
+
+Management resources and methods exist primarily to provision, configure, and
+audit the resources that the data plane interfaces with.
+
+As an example, the following are considered management resources for a cloud
+provider:
+
+- virtual machines
+- virtual private networks
+- disks
+- a blob store instance
+- a project or account
+
+The following behaviors are common (but not required) attributes of a
+management plane resource:
+
+- changes infrequently.
+
+### Data Plane
+
+Methods on the data plane operate on a variety of data formats, and generally
+interface with a resource provisioned via a management plane API. Examples of
+data plane methods include:
+
+- tables in a database
+- rows in a table
+- pushing or pulling from a message queue
+- uploading or downloading blobs from a blob store instance
+
+Data plane APIs **may** be heterogenous across a larger API surface, due to
+requirements including high throughput, low latency, or the need to adhere to an
+existing interface specification (e.g. ANSI SQL).
+
+- For convenience, resources and methods that operate on the data plane **may**
+expose themselves via resource-oriented management APIs. If so, those resources
+and methods **must** adhere to the requirements of the management plane.
+
+### Major distinctions between management and data plane
+
+- [IaC][] tools operate on the management plane exclusively.
+- Data planes provide much higher reliability, since they are often critical to
+  the runtime of applications.
+
+### Planes are contextual
+
+The definition of which plane a resource or operation resides on depends on the
+context. Therefore, the definition of management or data plane is incomplete
+without stating the context explicitly.
+
+For example, consider a cloud provider which exposes a Kubernetes cluster
+resource, which in turn exposes it's own API with resources. In the context of
+the cloud, the resources which live in the Kubernetes cluster are *data plane*.
+In the context of the Kubernetes cluster, the resources are *management plane*.
```

Comment:
When you say "resources" in here do you mean methods? b.c AIP-121 resources aren't "data plane". In some of the "management plane resources" above, they are all nouns, not verbs. So I just want a little clarity here.

Reply:
+1. This is about hosted 3P/OSS.

Reply:
removed the planes are contextual section entirely, will replace with a "1p vs 3p" discussion in the AIPs in a separate PR.